cloud.foundry.cli/src/main/java/cloud/foundry/cli/logic/apply/ServiceRequestsPlanner.java

```diff
@@ -16,156 +14,96 @@
 
-    /**
-     * Creates the requests for CfContainerChange.
-     *
-     * @param containerChange the CfContainerChange to be visited
-     * @throws ApplyException    if an error during the apply logic occurs. May
-     *                           contain another exception inside with more details.
-     * @throws IllegalArgumentException if the affected-object is not an instance of <code>serviceBean</code>.
-     */
-    @Override
-    public void visitContainerChange(CfContainerChange containerChange) {
-        if (this.getRequests().size() == 0) {
-            Object affectedObject = containerChange.getAffectedObject();
-            if (affectedObject instanceof ServiceBean) {
-                try {
-                    addUpdateServiceRequest((ServiceBean) affectedObject);
-                } catch (NullPointerException e) {
-                    throw new ApplyException(e);
-                }
-            } else {
-                throw new IllegalArgumentException("Only changes of services are permitted.");
-            }
-        }
-    }
+        if (hasNewObject(changes)) {
+            log.debug("Add create service request for service: " + serviceName);
 
-    /**
-     * Creates the requests for CfObjectValueChanged.
-     *
-     * @param objectValueChanged the CfObjectValueChanged to be visited
-     * @throws ApplyException    if an error during the apply logic occurs. May
-     *                           contain another exception inside with more details.
-     * @throws IllegalArgumentException if the affected-object is not an instance of <code>serviceBean</code>.
-     */
-    @Override
-    public void visitObjectValueChanged(CfObjectValueChanged objectValueChanged) {
-        if (this.getRequests().size() == 0) {
-            Object affectedObject = objectValueChanged.getAffectedObject();
-            if (affectedObject instanceof ServiceBean) {
-                try {
-                    addUpdateServiceRequest((ServiceBean) affectedObject);
-                } catch (NullPointerException e) {
-                    throw new ApplyException(e);
-                }
-            } else {
-                throw new IllegalArgumentException("Only changes of services are permitted.");
-            }
-        }
-    }
+            ServiceBean bean = (ServiceBean) getChange(changes, change -> change instanceof CfNewObject)
+                .get()
+                .getAffectedObject();
 
-    /**
-     * Creates requests for Update Service.
-     *
-     * @param affectedObject ServiceBean
-     */
-    private void addUpdateServiceRequest(ServiceBean affectedObject) {
-        this.addRequest(this.servicesOperations.update(serviceName, affectedObject));
-    }
+            return Flux.merge(this.servicesOperations.create(this.serviceName, bean));
+        } else if (hasRemovedObject(changes)) {
+            log.debug("Add remove service request for service: " + serviceName);
```

Comment:
Sorry, just saw it now. To be consistend with the logging Fabian has added to in application requests planner i would recommend to change this to.
```suggestion
            log.debug("Requesting removal of service", serviceName);
```